File: ch13/Chapter13-CS/App_Code/HelloSoapHeader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Services;
using System.Web.Services.Protocols;

/// <summary>
/// Summary description for HelloSoapHeader
/// </summary>
[WebService(Namespace = "http://tempuri.org/")]
[WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1, EmitConformanceClaims = true)]
public class HelloSoapHeader : System.Web.Services.WebService {

    public HelloHeader myHeader;

    [WebMethod]
    [SoapHeader("myHeader")]
    public string HelloWorld()
    {
        // Code here...
        if (myHeader == null)
        {
            return "Hello World";
        }
        else
        {
            return "Hello " + myHeader.Username + ". " +
               "<br>Your password is: " + myHeader.Password;
        }
    }
}

File: ch13/Chapter13-VB/App_Code/HelloSoapHeader.vb
Imports System.Web
Imports System.Web.Services
Imports System.Web.Services.Protocols

' To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line.
' <System.Web.Script.Services.ScriptService()> _
<WebService(Namespace:="http://tempuri.org/")> _
<WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1, EmitConformanceClaims:=True)> _
<Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _
Public Class HelloSoapHeader
    Inherits System.Web.Services.WebService

    Public myHeader As HelloHeader

    <WebMethod(), SoapHeader("myHeader")> _
    Public Function HelloWorld() As String
        If (myHeader Is Nothing) Then
            Return "Hello World"
        Else
            Return "Hello " & myHeader.Username & ". " & _
               "<br>Your password is: " & myHeader.Password
        End If
    End Function
End Class